module/product/src/Application/Validator/ProductHasChildren.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * Copyright © Ergonode Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Product\Application\Validator;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @Annotation
+ */
+class ProductHasChildren extends Constraint
+{
+    public string $message = 'Can\'t change binding attribute in product with variations.';
```

Comment:
```suggestion
    public string $message = 'Can\'t change binding attribute in product with variants.';
```

I don't think we use variation terminology anywhere?